java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/analyser/dependencies/RebuildService.java

```diff
@@ -0,0 +1,38 @@
+package com.redhat.hacbs.container.analyser.dependencies;
+
+import java.util.*;
+import java.util.function.UnaryOperator;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.logging.Log;
+
+@Singleton
+public class RebuildService {
+
+    @Inject
+    Instance<KubernetesClient> client;
+
+    public void rebuild(String taskRunName, Set<String> gavs) {
+
+        Log.infof("Identified %s Community Dependencies: %s", gavs.size(), new TreeSet<>(gavs));
+
+        var client = this.client.get();
+        var resources = client.resources(TaskRun.class);
+        resources.withName(taskRunName).editStatus(new UnaryOperator<TaskRun>() {
+            @Override
+            public TaskRun apply(TaskRun taskRun) {
+                List<TaskRunResult> results = new ArrayList<>();
+                if (taskRun.getStatus().getTaskResults() != null) {
+                    results.addAll(taskRun.getStatus().getTaskResults());
+                }
+                results.add(new TaskRunResult("java-community-dependencies", String.join(",", gavs)));
+                taskRun.getStatus().setTaskResults(results);
```

Comment:
It can't happen in the controller, the controller is in KCP land, the actual task is running in the compute cluster. We need some way to propagate this information back to the KCP control plane and at the moment all we have is results (without say creating a shared S3 account and sticking this in a bucket).

The controller does act on these results once they are back in the cluster, but it has no visibility into the compute cluster where the tasks are being run.

Reply:
yea per my comment below at https://github.com/redhat-appstudio/jvm-build-service/pull/272#discussion_r982525113 once I saw the golang code on the other side I course corrected and connected the dots